python-checks/src/main/java/org/sonar/python/checks/SkLearnEstimatorDontInitializeEstimatedValuesCheck.java

```diff
@@ -0,0 +1,157 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.quickfix.PythonQuickFix;
+import org.sonar.plugins.python.api.symbols.ClassSymbol;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.AssignmentStatement;
+import org.sonar.plugins.python.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.python.api.tree.ClassDef;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.Tuple;
+import org.sonar.python.quickfix.TextEditUtils;
+import org.sonar.python.tree.FunctionDefImpl;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6974")
+public class SkLearnEstimatorDontInitializeEstimatedValuesCheck extends PythonSubscriptionCheck {
+
+  private static final String BASE_ESTIMATOR_FULLY_QUALIFIED_NAME = "sklearn.base.BaseEstimator";
+  private static final Set<String> MIXINS_FULLY_QUALIFIED_NAME = Set.of(
+    "sklearn.base.BiclusterMixin",
+    "sklearn.base.ClassifierMixin",
+    "sklearn.base.ClusterMixin",
+    "sklearn.base.DensityMixin",
+    "sklearn.base.MetaEstimatorMixin",
+    "sklearn.base.OneToOneFeatureMixin",
+    "sklearn.base.OutlierMixin",
+    "sklearn.base.RegressorMixin",
+    "sklearn.base.TransformerMixin");
+
+  private static final String MESSAGE = "Move this estimated attribute in the `fit` method.";
+  private static final String MESSAGE_SECONDARY = "The attribute is used in this estimator";
+  public static final String QUICK_FIX_MESSAGE = "Remove the statement";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, SkLearnEstimatorDontInitializeEstimatedValuesCheck::checkFunction);
+  }
+
+  private static boolean inheritsMixin(ClassSymbol classSymbol) {
+    boolean directlyInheritsMixin = classSymbol.superClasses().stream().filter(superClass -> superClass.fullyQualifiedName() != null)
+      .anyMatch(superClass -> MIXINS_FULLY_QUALIFIED_NAME.contains(superClass.fullyQualifiedName()));
+    return directlyInheritsMixin || classSymbol.superClasses().stream().filter(symbol -> symbol.is(Symbol.Kind.CLASS)).map(ClassSymbol.class::cast)
+      .anyMatch(SkLearnEstimatorDontInitializeEstimatedValuesCheck::inheritsMixin);
```

Comment:
I think there is an easier way with the use of `ClassSymbol.isOrExtend(String fqn)` method.
```suggestion
    return MIXINS_FULLY_QUALIFIED_NAME.stream().anyMatch(mixin -> classSymbol.isOrExtend(mixin)); 
```
I did not test this so take it with a grain of salt :D

Reply:
It looks like it is working